index.js

```diff
@@ -498,12 +502,11 @@
   //. false
   //. ```
   function is(typeRep, x) {
-    if (_toString.call(typeRep.prototype['@@type']) === '[object String]') {
-      return x != null && x['@@type'] === typeRep.prototype['@@type'];
-    } else {
-      var match = /function (\w*)/.exec(typeRep);
```

Comment:
There actually *was* a good reason for this: `Function#name` is not specified by ES5. This change introduced a regression which was fortunately caught by the browser tests @Shard has hooked up in #298.

Mark, could you revert this change to `is` in #298? I believe it will fix some—perhaps all—of the test failures in Internet Explorer.